### ISSUE TITLE
Initial cormhap array jobs successful, but subsequent ones do not run

### ISSUE THREAD

**Issue by Ashleyfarlow** [marbl/canu#1263]
After experiencing similar [Meryl issues with 1.8 as others](https://github.com/marbl/canu/issues/1138)  I've downgraded to Canu 1.7.1 and successfully assemble the ecoli test data on PBSpro (with useGrid=True). 

However, when I attempt to assemble a human genome (about 40x) I can't make it past mhap in 1-overlapper.  

```
canu \
 -p ${sample} -d ${sample}.asm2 \
 genomeSize=3.1g \
 -pacbio-raw pacbio.raw.fastq/${sample}/*pacbio.raw.fq.gz \
 java=/apps/java/jdk1.8.0_60/bin/java \
 gridEngine="pbspro" gridOptions="-P gw26 -q biodev -r y" gridEngineThreadsOption="-l ncpus=THREADS,walltime=24:00:00" useGrid=True
```
note that `-r y` was essential to get allow submission of array jobs on our cluster.

The correction/2-correction folder is not generated. Within correction/1-overlapper it appears that the first 30 mhap array jobs ran successfully, cormhap_sampleID.o6088333.1 :  

```
Running job 1 based on PBS_ARRAY_INDEX=1 and offset=0.
Dumping reads from 1 to 48000 (inclusive).

Starting mhap precompute.

Running with these settings:
--filter-threshold = 5.0E-6
--help = false
--max-shift = 0.2
--min-olap-length = 500
--min-store-length = 0
--no-rc = false
--no-self = false
--no-tf = false
--num-hashes = 512
--num-min-matches = 3
--num-threads = 8
--ordered-kmer-size = 12
--ordered-sketch-size = 1536
--repeat-idf-scale = 10.0
--repeat-weight = 0.9
--settings = 0
--store-full-id = true
--supress-noise = 0
--threshold = 0.78
--version = false
-f = ../../0-mercounts/sampleid.ms16.frequentMers.ignore.gz
-h = false
-k = 16
-p = ./000001.input.fasta
-q = .
-s = 

Reading in filter file ../../0-mercounts/sampleid.ms16.frequentMers.ignore.gz.
Read in values for repeat 258 and 0
Warning, k-mer filter file has zero elements.
Initializing
Trying size 258 and it was successfull
Initialized
Time (s) to read filter file: 0.20703426800000002
Read in k-mer filter for sizes: [16]
Processing FASTA files for binary compression...
Current # sequences loaded and processed from file: 5000...
Current # sequences loaded and processed from file: 10000...
Current # sequences loaded and processed from file: 15000...
Current # sequences loaded and processed from file: 20000...
Current # sequences loaded and processed from file: 25000...
Current # sequences loaded and processed from file: 30000...
Current # sequences loaded and processed from file: 35000...
Current # sequences loaded and processed from file: 40000...
Current # sequences loaded and processed from file: 45000...
Current # sequences loaded and processed from file: 50000...
Current # sequences loaded and processed from file: 55000...
Current # sequences loaded and processed from file: 60000...
Current # sequences loaded and processed from file: 65000...
Current # sequences loaded and processed from file: 70000...
Current # sequences loaded and processed from file: 75000...
Current # sequences loaded and processed from file: 80000...
Current # sequences loaded and processed from file: 85000...
Current # sequences loaded and processed from file: 90000...
Current # sequences loaded and processed from file: 95000...
Processed 96000 sequences (fwd and rev).
Read, hashed, and stored file ./000001.input.fasta to ./000001.input.dat.
Time (s): 1727.0499099150002
Total time (s): 1727.0512260950002
```
but then there are several hundred subsequent array jobs with identical error logs:
```
Running job 1 based on PBS_ARRAY_INDEX=1 and offset=0.
Job previously completed successfully.
```
correction/1-overlapper also contains precompute.jobSubmit-01.out with only job ids:
```
6092340[].r-man2
```
these array job appear to have exited cleanly, `qstat -fx 6092340[].r-man2`

```
...
Resource_List.select = 1:ncpus=8:mpiprocs=8:mem=17179869184:job_tags=biodev:jobfs_local=524288000
Resource_List.walltime = 12:00:00
comment = Subjob finished
eligible_time = 11:08:59
Stageout_status = 1
Exit_status = 0
Submit_arguments = -j oe -l mem=16g -l ncpus=8,
	walltime=12:00:00 -P gw26 -q biodev -r y -N cormhap_sampleid -J 1-10 
	./precompute.sh
array = True
array_state_count = Queued:0 Running:0 Exiting:0 Expired:0 
array_indices_submitted = 1-10
...
```
and precompute.jobSubmit-01.sh:
```
#!/bin/sh

qsub -j oe \
  -l mem=16g -l ncpus=8,walltime=12:00:00 -P gw26 -q biodev -r y \
  -N "cormhap_sampleid" \
  -J 1-10 \
  ./precompute.sh  \
> ./precompute.jobSubmit-01.out 2>&1

```
there are also queries and blocks folders with data and queries.tar.

sampleid.report has no info beyond the kmer step.
canu.out:
```
-- Canu 1.7.1
--
-- CITATIONS
--
-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.
-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.
-- Genome Res. 2017 May;27(5):722-736.
-- http://doi.org/10.1101/gr.215087.116
-- 
-- Read and contig alignments during correction, consensus and GFA building use:
--   Šošic M, Šikic M.
--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.
--   Bioinformatics. 2017 May 1;33(9):1394-1395.
--   http://doi.org/10.1093/bioinformatics/btw753
-- 
-- Overlaps are generated using:
--   Berlin K, et al.
--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.
--   Nat Biotechnol. 2015 Jun;33(6):623-30.
--   http://doi.org/10.1038/nbt.3238
-- 
--   Myers EW, et al.
--   A Whole-Genome Assembly of Drosophila.
--   Science. 2000 Mar 24;287(5461):2196-204.
--   http://doi.org/10.1126/science.287.5461.2196
-- 
--   Li H.
--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.
--   Bioinformatics. 2016 Jul 15;32(14):2103-10.
--   http://doi.org/10.1093/bioinformatics/btw152
-- 
-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:
--   Chin CS, et al.
--   Phased diploid genome assembly with single-molecule real-time sequencing.
--   Nat Methods. 2016 Dec;13(12):1050-1054.
--   http://doi.org/10.1038/nmeth.4035
-- 
-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:
--   Chin CS, et al.
--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.
--   Nat Methods. 2013 Jun;10(6):563-9
--   http://doi.org/10.1038/nmeth.2474
-- 
-- CONFIGURE CANU
--
-- Detected Java(TM) Runtime Environment '1.8.0_60' (from '/apps/java/jdk1.8.0_60/bin/java').
-- Detected gnuplot version '4.2 patchlevel 6 ' (from 'gnuplot') and image format 'png'.
-- Detected 64 CPUs and 566 gigabytes of memory.
-- Detecting PBSPro resources.
-- 
-- Found   3 hosts with  32 cores and 3072 GB memory under PBSPro control.
-- Found  13 hosts with  24 cores and  256 GB memory under PBSPro control.
-- Found  12 hosts with  32 cores and  128 GB memory under PBSPro control.
-- Found   9 hosts with  32 cores and  576 GB memory under PBSPro control.
-- Found 2364 hosts with  16 cores and   32 GB memory under PBSPro control.
-- Found  10 hosts with  28 cores and 1024 GB memory under PBSPro control.
-- Found  19 hosts with  24 cores and  128 GB memory under PBSPro control.
-- Found 192 hosts with  32 cores and  192 GB memory under PBSPro control.
-- Found  72 hosts with  16 cores and  128 GB memory under PBSPro control.
-- Found 268 hosts with  28 cores and  256 GB memory under PBSPro control.
-- Found 1125 hosts with  16 cores and   64 GB memory under PBSPro control.
-- Found 532 hosts with  28 cores and  128 GB memory under PBSPro control.
-- Found  30 hosts with  64 cores and  192 GB memory under PBSPro control.
--
--                     (tag)Threads
--            (tag)Memory         |
--        (tag)         |         |  algorithm
--        -------  ------  --------  -----------------------------
-- Grid:  meryl    256 GB   32 CPUs  (k-mer counting)
-- Grid:  cormhap   16 GB    8 CPUs  (overlap detection with mhap)
-- Grid:  obtovl    24 GB   14 CPUs  (overlap detection)
-- Grid:  utgovl    24 GB   14 CPUs  (overlap detection)
-- Grid:  ovb        2 GB    1 CPU   (overlap store bucketizer)
-- Grid:  ovs       32 GB    1 CPU   (overlap store sorting)
-- Grid:  red        8 GB    4 CPUs  (read error detection)
-- Grid:  oea        8 GB    1 CPU   (overlap error adjustment)
-- Grid:  bat      512 GB   32 CPUs  (contig construction)
-- Grid:  gfa       32 GB   32 CPUs  (GFA alignment and processing)
--
-- In 'sampleid.gkpStore', found PacBio reads:
--   Raw:        8982414
--   Corrected:  0
--   Trimmed:    0
--
-- Generating assembly 'sampleid' in '/g/data1a/te53/longread.assembly/canu.assembly/sampleid/sampleid.asm2'
--
-- Parameters:
--
--  genomeSize        3100000000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.2400 ( 24.00%)
--    obtOvlErrorRate 0.0450 (  4.50%)
--    utgOvlErrorRate 0.0450 (  4.50%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.3000 ( 30.00%)
--    obtErrorRate    0.0450 (  4.50%)
--    utgErrorRate    0.0450 (  4.50%)
--    cnsErrorRate    0.0750 (  7.50%)
--
--
-- BEGIN CORRECTION
--
--
-- Mhap precompute jobs failed, tried 2 times, giving up.
--   job correction/1-overlapper/blocks/000011.dat FAILED.
--   job correction/1-overlapper/blocks/000012.dat FAILED.
--   job correction/1-overlapper/blocks/000013.dat FAILED.
--   job correction/1-overlapper/blocks/000014.dat FAILED.
--   job correction/1-overlapper/blocks/000015.dat FAILED.
--   job correction/1-overlapper/blocks/000016.dat FAILED.
--   job correction/1-overlapper/blocks/000017.dat FAILED.
--   job correction/1-overlapper/blocks/000018.dat FAILED.
--   job correction/1-overlapper/blocks/000019.dat FAILED.
--   job correction/1-overlapper/blocks/000020.dat FAILED.
--   job correction/1-overlapper/blocks/000021.dat FAILED.
--   job correction/1-overlapper/blocks/000022.dat FAILED.
--   job correction/1-overlapper/blocks/000023.dat FAILED.
--   job correction/1-overlapper/blocks/000024.dat FAILED.
--   job correction/1-overlapper/blocks/000025.dat FAILED.
--   job correction/1-overlapper/blocks/000026.dat FAILED.
--   job correction/1-overlapper/blocks/000027.dat FAILED.
--   job correction/1-overlapper/blocks/000028.dat FAILED.
--   job correction/1-overlapper/blocks/000029.dat FAILED.
--   job correction/1-overlapper/blocks/000030.dat FAILED.
--   job correction/1-overlapper/blocks/000031.dat FAILED.
--   job correction/1-overlapper/blocks/000032.dat FAILED.
--   job correction/1-overlapper/blocks/000033.dat FAILED.
--   job correction/1-overlapper/blocks/000034.dat FAILED.
--   job correction/1-overlapper/blocks/000035.dat FAILED.
--   job correction/1-overlapper/blocks/000036.dat FAILED.
--   job correction/1-overlapper/blocks/000037.dat FAILED.
--   job correction/1-overlapper/blocks/000038.dat FAILED.
--   job correction/1-overlapper/blocks/000039.dat FAILED.
--   job correction/1-overlapper/blocks/000040.dat FAILED.
--   job correction/1-overlapper/blocks/000041.dat FAILED.
--   job correction/1-overlapper/blocks/000042.dat FAILED.
--   job correction/1-overlapper/blocks/000043.dat FAILED.
--   job correction/1-overlapper/blocks/000044.dat FAILED.
--   job correction/1-overlapper/blocks/000045.dat FAILED.
--   job correction/1-overlapper/blocks/000046.dat FAILED.
--   job correction/1-overlapper/blocks/000047.dat FAILED.
--   job correction/1-overlapper/blocks/000048.dat FAILED.
--   job correction/1-overlapper/blocks/000049.dat FAILED.
--   job correction/1-overlapper/blocks/000050.dat FAILED.
--   job correction/1-overlapper/blocks/000051.dat FAILED.
--   job correction/1-overlapper/blocks/000052.dat FAILED.
--   job correction/1-overlapper/blocks/000053.dat FAILED.
--   job correction/1-overlapper/blocks/000054.dat FAILED.
--   job correction/1-overlapper/blocks/000055.dat FAILED.
--   job correction/1-overlapper/blocks/000056.dat FAILED.
--   job correction/1-overlapper/blocks/000057.dat FAILED.
--   job correction/1-overlapper/blocks/000058.dat FAILED.
--   job correction/1-overlapper/blocks/000059.dat FAILED.
--   job correction/1-overlapper/blocks/000060.dat FAILED.
--   job correction/1-overlapper/blocks/000061.dat FAILED.
--   job correction/1-overlapper/blocks/000062.dat FAILED.
--   job correction/1-overlapper/blocks/000063.dat FAILED.
--   job correction/1-overlapper/blocks/000064.dat FAILED.
--   job correction/1-overlapper/blocks/000065.dat FAILED.
--   job correction/1-overlapper/blocks/000066.dat FAILED.
--   job correction/1-overlapper/blocks/000067.dat FAILED.
--   job correction/1-overlapper/blocks/000068.dat FAILED.
--   job correction/1-overlapper/blocks/000069.dat FAILED.
--   job correction/1-overlapper/blocks/000070.dat FAILED.
--   job correction/1-overlapper/blocks/000071.dat FAILED.
--   job correction/1-overlapper/blocks/000072.dat FAILED.
--   job correction/1-overlapper/blocks/000073.dat FAILED.
--   job correction/1-overlapper/blocks/000074.dat FAILED.
--   job correction/1-overlapper/blocks/000075.dat FAILED.
--   job correction/1-overlapper/blocks/000076.dat FAILED.
--   job correction/1-overlapper/blocks/000077.dat FAILED.
--   job correction/1-overlapper/blocks/000078.dat FAILED.
--   job correction/1-overlapper/blocks/000079.dat FAILED.
--   job correction/1-overlapper/blocks/000080.dat FAILED.
--   job correction/1-overlapper/blocks/000081.dat FAILED.
--   job correction/1-overlapper/blocks/000082.dat FAILED.
--   job correction/1-overlapper/blocks/000083.dat FAILED.
--   job correction/1-overlapper/blocks/000084.dat FAILED.
--   job correction/1-overlapper/blocks/000085.dat FAILED.
--   job correction/1-overlapper/blocks/000086.dat FAILED.
--   job correction/1-overlapper/blocks/000087.dat FAILED.
--   job correction/1-overlapper/blocks/000088.dat FAILED.
--   job correction/1-overlapper/blocks/000089.dat FAILED.
--   job correction/1-overlapper/blocks/000090.dat FAILED.
--   job correction/1-overlapper/blocks/000091.dat FAILED.
--   job correction/1-overlapper/blocks/000092.dat FAILED.
--   job correction/1-overlapper/blocks/000093.dat FAILED.
--   job correction/1-overlapper/blocks/000094.dat FAILED.
--   job correction/1-overlapper/blocks/000095.dat FAILED.
--   job correction/1-overlapper/blocks/000096.dat FAILED.
--   job correction/1-overlapper/blocks/000097.dat FAILED.
--   job correction/1-overlapper/blocks/000098.dat FAILED.
--   job correction/1-overlapper/blocks/000099.dat FAILED.
--   job correction/1-overlapper/blocks/000100.dat FAILED.
--   job correction/1-overlapper/blocks/000101.dat FAILED.
--   job correction/1-overlapper/blocks/000102.dat FAILED.
--   job correction/1-overlapper/blocks/000103.dat FAILED.
--   job correction/1-overlapper/blocks/000104.dat FAILED.
--   job correction/1-overlapper/blocks/000105.dat FAILED.
--   job correction/1-overlapper/blocks/000106.dat FAILED.
--   job correction/1-overlapper/blocks/000107.dat FAILED.
--   job correction/1-overlapper/blocks/000108.dat FAILED.
--   job correction/1-overlapper/blocks/000109.dat FAILED.
--   job correction/1-overlapper/blocks/000110.dat FAILED.
--   job correction/1-overlapper/blocks/000111.dat FAILED.
--   job correction/1-overlapper/blocks/000112.dat FAILED.
--   job correction/1-overlapper/blocks/000113.dat FAILED.
--   job correction/1-overlapper/blocks/000114.dat FAILED.
--   job correction/1-overlapper/blocks/000115.dat FAILED.
--   job correction/1-overlapper/blocks/000116.dat FAILED.
--   job correction/1-overlapper/blocks/000117.dat FAILED.
--   job correction/1-overlapper/blocks/000118.dat FAILED.
--   job correction/1-overlapper/blocks/000119.dat FAILED.
--   job correction/1-overlapper/blocks/000120.dat FAILED.
--   job correction/1-overlapper/blocks/000121.dat FAILED.
--   job correction/1-overlapper/blocks/000122.dat FAILED.
--   job correction/1-overlapper/blocks/000123.dat FAILED.
--   job correction/1-overlapper/blocks/000124.dat FAILED.
--   job correction/1-overlapper/blocks/000125.dat FAILED.
--   job correction/1-overlapper/blocks/000126.dat FAILED.
--   job correction/1-overlapper/blocks/000127.dat FAILED.
--   job correction/1-overlapper/blocks/000128.dat FAILED.
--   job correction/1-overlapper/blocks/000129.dat FAILED.
--   job correction/1-overlapper/blocks/000130.dat FAILED.
--   job correction/1-overlapper/blocks/000131.dat FAILED.
--   job correction/1-overlapper/blocks/000132.dat FAILED.
--   job correction/1-overlapper/blocks/000133.dat FAILED.
--   job correction/1-overlapper/blocks/000134.dat FAILED.
--   job correction/1-overlapper/blocks/000135.dat FAILED.
--   job correction/1-overlapper/blocks/000136.dat FAILED.
--   job correction/1-overlapper/blocks/000137.dat FAILED.
--   job correction/1-overlapper/blocks/000138.dat FAILED.
--   job correction/1-overlapper/blocks/000139.dat FAILED.
--   job correction/1-overlapper/blocks/000140.dat FAILED.
--   job correction/1-overlapper/blocks/000141.dat FAILED.
--   job correction/1-overlapper/blocks/000142.dat FAILED.
--   job correction/1-overlapper/blocks/000143.dat FAILED.
--   job correction/1-overlapper/blocks/000144.dat FAILED.
--   job correction/1-overlapper/blocks/000145.dat FAILED.
--   job correction/1-overlapper/blocks/000146.dat FAILED.
--   job correction/1-overlapper/blocks/000147.dat FAILED.
--   job correction/1-overlapper/blocks/000148.dat FAILED.
--   job correction/1-overlapper/blocks/000149.dat FAILED.
--   job correction/1-overlapper/blocks/000150.dat FAILED.
--   job correction/1-overlapper/blocks/000151.dat FAILED.
--   job correction/1-overlapper/blocks/000152.dat FAILED.
--   job correction/1-overlapper/blocks/000153.dat FAILED.
--   job correction/1-overlapper/blocks/000154.dat FAILED.
--   job correction/1-overlapper/blocks/000155.dat FAILED.
--   job correction/1-overlapper/blocks/000156.dat FAILED.
--   job correction/1-overlapper/blocks/000157.dat FAILED.
--   job correction/1-overlapper/blocks/000158.dat FAILED.
--   job correction/1-overlapper/blocks/000159.dat FAILED.
--   job correction/1-overlapper/blocks/000160.dat FAILED.
--   job correction/1-overlapper/blocks/000161.dat FAILED.
--   job correction/1-overlapper/blocks/000162.dat FAILED.
--   job correction/1-overlapper/blocks/000163.dat FAILED.
--   job correction/1-overlapper/blocks/000164.dat FAILED.
--   job correction/1-overlapper/blocks/000165.dat FAILED.
--   job correction/1-overlapper/blocks/000166.dat FAILED.
--   job correction/1-overlapper/blocks/000167.dat FAILED.
--   job correction/1-overlapper/blocks/000168.dat FAILED.
--   job correction/1-overlapper/blocks/000169.dat FAILED.
--   job correction/1-overlapper/blocks/000170.dat FAILED.
--   job correction/1-overlapper/blocks/000171.dat FAILED.
--   job correction/1-overlapper/blocks/000172.dat FAILED.
--   job correction/1-overlapper/blocks/000173.dat FAILED.
--   job correction/1-overlapper/blocks/000174.dat FAILED.
--   job correction/1-overlapper/blocks/000175.dat FAILED.
--   job correction/1-overlapper/blocks/000176.dat FAILED.
--   job correction/1-overlapper/blocks/000177.dat FAILED.
--   job correction/1-overlapper/blocks/000178.dat FAILED.
--   job correction/1-overlapper/blocks/000179.dat FAILED.
--   job correction/1-overlapper/blocks/000180.dat FAILED.
--   job correction/1-overlapper/blocks/000181.dat FAILED.
--   job correction/1-overlapper/blocks/000182.dat FAILED.
--   job correction/1-overlapper/blocks/000183.dat FAILED.
--   job correction/1-overlapper/blocks/000184.dat FAILED.
--   job correction/1-overlapper/blocks/000185.dat FAILED.
--   job correction/1-overlapper/blocks/000186.dat FAILED.
--   job correction/1-overlapper/blocks/000187.dat FAILED.
--   job correction/1-overlapper/blocks/000188.dat FAILED.
--

ABORT:
ABORT: Canu 1.7.1
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:

======================================================================================
                  Resource Usage on 2019-02-25 00:07:47:
   Job Id:             6092358.r-man2
   Project:            gw26
   Exit Status:        1
   Service Units:      0.01
   NCPUs Requested:    1                      NCPUs Used: 1               
                                           CPU Time Used: 00:00:01                                   
   Memory Requested:   4.0GB                 Memory Used: 9.43MB          
   Walltime requested: 12:00:00            Walltime Used: 00:00:20        
   JobFS requested:    500.0MB                JobFS used: 0B              
======================================================================================

```

**Comment by skoren** [comment]
This is likely a PBS issue, we don't have access to a PBSPro system to test those changes. There were fixes to PBS support between 1.7 and 1.8 so by downgrading it looks like you're encountering one of those old issues. 

It seems your grid is reporting it only supports 10 job arrays, is that accurate? What does ```qmgr -c 'p s' ``` report? One of the fixes was to properly pass the offset for jobs > this max (11 and greater in your case) so in your version the offset isn't being passed and so job 11 thinks it is job 1 as does job 21, 31, and so on. If your grid supports more than 10 job arrays, you should be able to just tell canu to create one large job with ```gridEngineArrayMaxJobs=65000``` or whatever your limit is if one exists. If it really only supports 10-job arrays then you'd need to edit the code to apply the fix from 1.8 to 1.7.

**Comment by Ashleyfarlow** [comment]
Thanks Sergey, you are correct:
```
set server max_array_size = 10
set queue biodev max_queued += [p:gw26=1000]
```
I've set `gridEngineArrayMaxJobs=10` , have I interpreted this correctly? And is this going to slow down the assembly? 

Clearly it's preferable that I get 1.8 working. When running on the ecoli test data things stop in `ecoli-pacbio/correction/0-mercounts/`  

 meryl-count.jobSubmit-01.sh:
```
#!/bin/sh

qsub -j oe \
  -l mem=8g -l ncpus=4,walltime=2:00:00 -P te53 -q express \
  -N "meryl_ecoli" \
   \
 --   ./meryl-count.sh 0 \
> ./meryl-count.jobSubmit-01.out 2>&1
```

meryl-count.jobSubmit-01.out 
```
6037330.r-man2
```

meryl_ecoli.o6037330:
```
ERROR: Could not execv ./meryl-count.sh! ret=-1 errno=2
```

meryl-configure.err:
```
Found perl:
   /short/te53/af3298/miniconda3/bin/perl
   This is perl 5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi

Found java:
   /apps/java/jdk1.8.0_60/bin/java
   java version "1.8.0_60"

Found canu:
   /short/te53/software/canu/canu-1.8/Linux-amd64/bin/canu
   Canu 1.8
```

Any ideas?

**Comment by skoren** [comment]
Setting it to 10 won't help since that's the default it is already picking up. Is this accurate, are arrays really limited to 10 jobs on your system? I would suggest setting ```gridEngineArrayMaxJobs=2000``` and see if that runs, this will avoid using the offset which is hitting a bug.

Otherwise, you'd need to edit the code in src/pipelines/canu/Execution.pm from:
```
 718         $off = "";
```
to
```
718        $off = $bgn if (($bgn == $end) && ($opt =~ m/ARRAY_JOBS/));
```

**Comment by Ashleyfarlow** [comment]
Thanks again. I've still having problems. I followed your advice above and included gridEngineArrayMaxJobs=2000 and also made the recommended change to src/pipelines/canu/Execution.pm however, precompute.jobSubmit-01.sh is still atempting to submit >10 array jobs:  
```
#!/bin/sh

qsub -j oe \
  -l mem=16g -l ncpus=8,walltime=24:00:00 -P te53 -q express -r y \
  -N "cormhap_ELD102591" \
  -J 1-188 \
  ./precompute.sh  \
> ./precompute.jobSubmit-01.out 2>&1
```
I've spoken with our HPC support and `max_array_size = 10` is truly at hard limit that they are unwilling to change and I quote `The 10 sub-job limit for array jobs on Raijin is a server wide limit that we won't change.
There are a number of significant limitations in the way that sub-jobs are processed for array jobs in PBS Pro that we really recommend not using them at all. One of the biggest ones that would impact you is if for any reason the PBS server process were to be restarted (or it crashed) all the sub-jobs are forgotten and re-created, causing them all to be re-run. There are a number of other limitations around the ability to efficiently manage the jobs that also come into things, which can reduce the throughput of jobs both for the array job, and for other jobs on the system.`
Their rather impractical advice was for me to recode canu placing the qsub command within a for look to "fake" a array job. 

Is there a way I can avoid array jobs all together? any other suggestions?

**Comment by Ashleyfarlow** [comment]
I've bypassed precompute.jobSubmit-01.sh with 
```
#!/bin/bash
for i in $(seq 1 188); do
  sleep .2
  qsub -j oe \
  -l mem=16g -l ncpus=8,walltime=24:00:00 -P te53 -q express -r y -l wd \
  -N "cormhap_sampleID" \
  -v PBS_ARRAY_INDEX=$i \
  ./precompute.sh  \
>> ./precompute.jobSubmit-01.out 2>&1
done
```
 do you think this is reasonable? and will I face the same issue (-J >10 ) at subsequent steps?

**Comment by skoren** [comment]
Yes, the way canu works around the 10 job limit is to submit multiple 10-job arrays each with an offset. So the first array has an offset of 0, the second an offset of 10, etc. You don't need both gridEngineArrayMaxJobs and the bug fix. The bug fix (which stops canu from losing the offset) should let canu run with gridEngineArrayMaxJobs=10 successfully.

**Comment by skoren** [comment]
Any updates, did Canu run with the code change thorough the next steps?

**Comment by skoren** [comment]
Idle, should be fixed by code change in 1.8 + PBS meryl fixes.
